### PR TITLE
Fix goroutine leak in socket.sender 

### PR DESCRIPTION
**Pull request by davidmirza408**
When socket.sender is waiting to send a message to a socket that has been closed, it will wait indefinitely on s.cv.Wait().  This change sends a signal to notify the s.cv.wait called in socket.sender that the socket has been closed.

### CODE REVIEW

**Comment by davidmirza408** [comment]
mistake

**Comment by davidmirza408** [comment]
@gdamore can you review my PR?

File: protocol/xpush/xpush.go [hunk 1]
```diff
@@ -261,6 +261,7 @@ func (s *socket) Close() error {
 		return protocol.ErrClosed
 	}
 	s.closed = true
+	s.cv.Signal()
```
Comment (gdamore) [inline protocol/xpush/xpush.go:264]: This should be s.cv.Broadcast() (because what if multiple threads are waiting to send?)

**Review by gdamore — CHANGES_REQUESTED** [review]

**Comment by davidmirza408** [comment]
@gdamore thanks for the review.  I updated the PR.

**Review by gdamore — APPROVED** [review]

**Pull request merged**